lib/chef/provisioning/chef_provider_action_handler.rb

```diff
@@ -42,16 +42,23 @@ def report_progress(description)
       provider.converge_by description do
         # We already did the action, but we trust whoever told us that they did it.
       end
+      puts "\n"
     end
 
     def performed_action(description)
       provider.converge_by description do
         # We already did the action, but we trust whoever told us that they did it.
       end
+      puts "\n"
```

Comment:
`converge_by` is filtering through the chef event formatter just fine, but there are[other events](https://github.com/chef/chef/blob/master/lib/chef/formatters/doc.rb#L277-L281) that are NOT being called.  In the normal flow, `converge_by` only calls (resource_update_applied)[https://github.com/chef/chef/blob/master/lib/chef/formatters/doc.rb#L259-L274] via [why_run](https://github.com/chef/chef/blob/master/lib/chef/mixin/why_run.rb#L54).

If we have access to the `@events` bus from the run_context, we could call `resource_updated` ourselves.  But that feels pretty hacky to me - `provider.run_context.events.resource_updated(provider.new_resource, provider.action)`.

What about calling `provider.run_action` instead of `provider.converge_by`?  That would perform all the correct event formatting (I think...).